smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ImportDeclarations.java

```diff
@@ -39,21 +41,25 @@
         this.relativize = Paths.get(relativize).getParent();
     }
 
+    ImportDeclarations addDefaultImport(String name, String module) {
+        module = getRelativizedModule(relativize, module);
+
+        if (!module.isEmpty() && (relativize == null || !module.equals(relativize.toString()))) {
```

Comment:
Can simplify to:

```
if (!module.isEmpty() && Objects.equals(relativize, module)) {
```

Reply:
The suggestion shows warning `"String seems to be unrelated to Path"` as the types of module (String) and relativize (Path) are different
<details>
<summary>Screenshot</summary>

![Screen Shot 2020-09-11 at 10 46 10 AM](https://user-images.githubusercontent.com/16024985/92956624-0a762d00-f41c-11ea-8d27-88c0fceb5c1a.png)

</details>

I can't call `Objects.equals(relativize.toString(), module)` as that removes null check.
Calling `Objects.equals(relativize, Paths.get(module))` fails multiple unit tests